### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by okhlif**
# Description

- Fix vbr performance : Fix 1st pass bug from rebasing the PR #1457
- Fix vbr performance : Fix 1st pass bug in 10bit
- Fix RC token check to include double dash
- Adding the missing flag from previous commits RC_BUG_FIXES (Fix the one pass QP assignment using frames_to_be_encoded)

# Issue

# Author(s)

@okhlif 
@anaghdin 
# Performance impact

- [x] quality
- [ ] memory
- [ ] speed
- [ ] 8 bit
- [ ] 10 bit
- [ ] N/A

# Test set
- [ ] obj-1-fast can be found [here](https://media.xiph.org/video/derf/objective-1-fast.tar.gz)
- [x] other
- [ ] N/A

# Merge method
- [x] Allow the maintainer to squash and merge when PR is ready to create a 1-commit to the master branch. The maintainer will be able to fix typos / combine commit messages to create a more readable 1-commit message or use whatever is stated in the 'Description' section
- [ ] I will clean up my commits and the maintainer shall use 'rebase and merge' to the master branch

### CODE REVIEW

File: Source/API/EbDebugMacros.h [hunk 1]
```diff
@@ -35,6 +35,7 @@ extern "C" {
 #define NON_AVX512_SUPPORT
 #endif
 
+#define RC_BUG_FIXES 1 // Fix the one pass QP assignment using frames_to_be_encoded
```
Comment (hassount) [inline Source/API/EbDebugMacros.h:38]: @okhlif thank you for the PR, can you please follow the nomenclature guidelines as shown in lines 14-23 above?

Comment (okhlif) [reply Source/API/EbDebugMacros.h]: Done, Thank you

**Comment by hassount** [comment]
LGTM, Pending testing to pass from @OmranAbdelkafi

**Comment by ccccheung** [comment]
Tested with objective 1 fast 8 bit and 10 bit
Here are the results for 2 pass VBR vs 1 pass CRF with this PR

  | BD-Rate   Deviation 
-- | --
M0 - 8bit| 16.26%
M4 - 8bit| 7.12%
M8 - 8bit| 6.20%
M0 - 10bit| 73.64%
M4 - 10bit| 7.42%
M8 - 10bit| 6.44%

**Comment by ccccheung** [comment]
Tested 2 pass VBR vs 1 pass CRF with this PR using a
low resolution test set
  | Speed Deviation  | BD-Rate   Deviation | Memory Deviation
-- | -- | -- | --
M0 - 8bit|  -1.32%  | 3.21% | 9.11%
M4 - 8bit|  -2.27%  | 3.07% | 9.17%
M8 - 8bit|  -11.38% | 2.99% | 10.02%

**Pull request merged**